src/cometh/rental-protocol/rental-ended-handler.ts

```diff
@@ -0,0 +1,71 @@
+import { log } from "@graphprotocol/graph-ts";
+import { Nft, Rental } from "../../../generated/schema";
+import { generateNftId } from "../../utils/misc";
+import { COMETHSPACESHIP } from "../../utils/constants";
+import { RentalEnded } from "../../../generated/ComethRentalProtocol/ComethRentalProtocol";
+import { SPACESHIP_ADDRESS } from "../../utils/addresses";
+
+/**
+ *     event RentalEnded(
+ *       address indexed lender,
+ *       address indexed tenant,
+ *       address token,
+ *       uint256 tokenId
+ *   );
+ */
+export function handleRentalEnded(event: RentalEnded): void {
+  if (event.params.token.toHexString() != SPACESHIP_ADDRESS) {
+    log.debug(
+      "[handleRentalEnded] NFT {} is not a spaceship, tx: {}, skipping...",
+      [event.params.tokenId.toString(), event.transaction.hash.toHex()]
+    );
+    return;
+  }
+
+  const nftId = generateNftId(COMETHSPACESHIP, event.params.tokenId);
+
+  const nft = Nft.load(nftId);
+  if (!nft) {
+    log.debug("[handleRentalEnded] Spaceship {} does not exist, tx: {}", [
+      event.params.tokenId.toString(),
+      event.transaction.hash.toHex(),
+    ]);
+    return;
+  }
+
+  const currentRentalId = nft.currentRental;
+
+  if (!currentRentalId) {
+    log.warning(
+      "[handleRentalEnded] NFT {} has no rental, tx: {}, skipping ...",
+      [nft.id, event.transaction.hash.toHex()]
+    );
+    return;
+  }
+
+  const currentRental = Rental.load(currentRentalId!);
+
+  if (!currentRental) {
+    throw new Error(
+      "[handleRentalEnded] Rental " +
+        currentRentalId! +
+        " does not exist, tx: " +
+        event.transaction.hash.toHex()
+    );
+  }
+
+  // update rental
+  currentRental.expiration_date = event.block.timestamp;
```

Comment:
The expiration date is not the same as the date when the Rental ended. We need an additional field for this `expiration_date` should be set when creating the Rental, and points to when the Rental expires, and `ended_at` can be the date when the Rental ends (and should be set on this handler)